### PR TITLE
🔬 Add eslint support for Typescript

### PR DESCRIPTION
**Pull request by thibmaek**
**What**: Adds support so Typescript (.ts) files can be linted

**Why**: Linting Typescript files does not work

**How**: Added the dep. [`typescript-eslint-parser`](https://github.com/eslint/typescript-eslint-parser) and checked if it is in project using ifAnyDep.

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged 
- [x] Added myself to contributors table

### CODE REVIEW

File: src/config/eslintrc.js [hunk 1]
```diff
@@ -11,7 +11,11 @@ module.exports = {
     ifAnyDep('flow-bin', 'eslint-config-prettier/flowtype'),
     require.resolve('eslint-config-prettier/react'),
   ].filter(Boolean),
-  parser: ifAnyDep(['babel-cli', 'babel-core', 'babel-eslint', 'babel-loader'], 'babel-eslint'),
+  parser: ifAnyDep(
+    ['babel-cli', 'babel-core', 'babel-eslint', 'babel-loader'],
+    'babel-eslint',
+    ifAnyDep(['typescript'], 'typescript-eslint-parser')
```
Comment (easybird) [inline src/config/eslintrc.js]: shouldn't this be the other way around?
```

ifAnyDep(['typescript'], 

(THEN) 'typescript-eslint-parser', 

(ELSE) ifAnyDep(['babel-cli', 'babel-core', 'babel-eslint', 'babel-loader'], 'babel-eslint')
```

Comment (easybird) [reply src/config/eslintrc.js]: but if a project has a dependendy on both babel-cli and typescript, in this case, the typescript-eslint-parser won't be installed. Is this expected behavior?

Comment (thibmaek) [reply src/config/eslintrc.js]: Could be argued about but I don't think you should have Babel transpile code when you already have Typescript doing it for you, at least that's how the TS project implementing itp-react-script currently rolls.

Don't really see why you would include both.

Comment (easybird) [reply src/config/eslintrc.js]: ok, good for me!

**Pull request merged**